src/lib/apiTypesV2/Subscription.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+#define SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <string>
+#include <vector>
+
+#include "../ngsi/Duration.h"
+#include "../ngsi/Throttling.h"
+
+struct EntID
+{
+  std::string id;
+  std::string idPattern;
+  std::string type;
+  std::string toJson();
+};
+
+struct Notification
+{
+  std::vector<std::string> attributes;
+  std::string callback;
+  Throttling throttling;
+  std::string toJson();
+};
+
+struct Condition
+{
+  std::vector<std::string> attributes;
+  struct {
+    std::string q;
+    std::string geometry;
+    std::string coords;
+  } expression;
+  std::string toJson();
+};
+
+struct Subject
+{
+  std::vector<EntID> entities;
+  Condition condition;
+  std::string toJson();
+};
+
+struct Subscription
+{
+  std::string id;
+  Subject subject;
+  Duration duration;
+  Notification notification;
+  std::string toJson();
+};
```

Comment:
I see two options here.
1. If these classes are only to be used for json rendering of the subscription list it can all stay the way it is. I would change the name of the classes though, to reflect what they actually do.
It would be clearer if the classes were a part of the Subscription class, though.

2. If the classes are to be used for something more than only rendering of JSON, they must be includable separately. If somebody needs the class Subject (just an example) but doesn't want the class Notification, that somebody should be able to get only Subject. Perhaps that include already has a class called Notification ...
This implies a separate file for each class.